src/Factory/ResourceControllerFactory.php

```diff
@@ -75,17 +74,15 @@ public function canCreateServiceWithName(ServiceLocatorInterface $controllers, $
     /**
      * Create service with name
      *
-     * @param \Zend\ServiceManager\AbstractPluginManager $controllers
-     * @param string                  $name
      * @param string                  $requestedName
      *
      * @return ResourceController
      *
      * @throws ServiceNotCreatedException if listener specified is not a ListenerAggregate
      */
-    public function createServiceWithName(ServiceLocatorInterface $controllers, $name, $requestedName)
+    public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null)
     {
-        $services = $controllers->getServiceLocator();
+        $services = $container;
```

Comment:
These changes are what Zend recommended for the upgrade path here (was in the docblocks of the old interface this implemented).